repo/fsrepo/lock.go

```diff
@@ -0,0 +1,54 @@
+package fsrepo
+
+import (
+	"path"
+	"sync"
+)
+
+type packageLock struct {
+	// lock protects repos
+	lock sync.Mutex
+	// repos maps repo paths to the number of openers holding an FSRepo handle
+	// to it
+	repos map[string]int
+}
+
+func makePackageLock() *packageLock {
+	return &packageLock{
+		repos: make(map[string]int),
+	}
+}
```

Comment:
instead of needing to make it with `init`, you can make the zero value be ready to be used like so:

```
func (l *packageLock) Lock() {
  l.lock.Lock()
  if l.repos == nil {
    l.repos = make(map[string]int)
  }
}